File: CrossFiles/Java/Class2.java
import java.sql.*;

public class Class2 {
    private String input;
    private Connection conn;

    public Class2(String input) {
        this.input = input;
    }

    public void connectDb(String dbUrl) {
        try {
            conn = DriverManager.getConnection(dbUrl);
        } catch (SQLException e) {
            System.out.println("Connection failed: " + e.getMessage());
        }
    }

    public void process() {
        System.out.println("Class2 processing: " + input);
        if (conn == null) {
            System.out.println("Database not connected");
            return;
        }

        // Potentially unsafe operation
        String query = "SELECT * FROM users WHERE username = '" + input + "'";
        System.out.println("Executing query: " + query);

        try {
            Statement stmt = conn.createStatement();
            ResultSet rs = stmt.executeQuery(query);
            while (rs.next()) {
                System.out.println(rs.getString("username"));
            }
        } catch (SQLException e) {
            System.out.println("Query failed: " + e.getMessage());
        }
    }

    public void closeDb() {
        try {
            if (conn != null) {
                conn.close();
            }
        } catch (SQLException e) {
            System.out.println("Error closing connection: " + e.getMessage());
        }
    }
}

File: CrossFiles/C++/Class2.cpp
#include "Class2.h"
#include <iostream>

Class2::Class2(const std::string& input) : input(input), db(nullptr) {}

Class2::~Class2() {
    if (db) {
        sqlite3_close(db);
    }
}

bool Class2::connectDb(const std::string& dbName) {
    int rc = sqlite3_open(dbName.c_str(), &db);
    if (rc) {
        std::cerr << "Can't open database: " << sqlite3_errmsg(db) << std::endl;
        return false;
    }
    return true;
}

void Class2::process() {
    std::cout << "Class2 processing: " << input << std::endl;
    if (!db) {
        std::cout << "Database not connected" << std::endl;
        return;
    }

    // Potentially unsafe operation
    std::string query = "SELECT * FROM users WHERE username = '" + input + "'";
    std::cout << "Executing query: " << query << std::endl;

    char* errMsg = nullptr;
    int rc = sqlite3_exec(db, query.c_str(), nullptr, nullptr, &errMsg);
    if (rc != SQLITE_OK) {
        std::cerr << "SQL error: " << errMsg << std::endl;
        sqlite3_free(errMsg);
    } else {
        std::cout << "Operation done successfully" << std::endl;
    }
}

